### DISCUSSION TITLE
[Feature request] Add support for PhpRedis Redis::SERIALIZER_IGBINARY and Redis::COMPRESSION_LZ4

### DISCUSSION THREAD

**Discussion by cyppe** [beyondcode/herd-community#470] [Ideas]
I am using the following serializer and compression from phpredis, as documented here:

https://laravel.com/docs/11.x/redis#phpredis-serialization

```
        'options' => [
            'cluster'     => env('REDIS_CLUSTER', 'redis'),
            'prefix'      => env('REDIS_PREFIX', Str::slug(env('APP_NAME', 'laravel'), '_') . '_database_'),
            'serializer'  => Redis::SERIALIZER_IGBINARY,
            'compression' => Redis::COMPRESSION_LZ4,
        ],

```

It would be really nice if igbinary and lz4 would be added. And to make it a bit more broadly usable probably add the other serializers and compressions supported by PhpRedis. But this combo is the most performant from my tests.

For what it's worth they exists on Forge out of the box if the goal is to be 1:1 compatible with Forge standard as it seems in other discussions about some services versions.  

I am on Pro license.

**Comment by cyppe** [comment]
I tried to build it myself even though I suspected redis was compiled in and not loaded as module.

And unfortunately that was correct assumption.

Built them and loading:

extension=/opt/homebrew/lib/php/pecl/20230831/igbinary.so
extension=/opt/homebrew/lib/php/pecl/20230831/redis.so


igbinary works just fine, no surprise there.

But redis does not:

❯ php --ri redis

**Warning: Module "redis" is already loaded in Unknown on line 0**


Wrote down a few stats how to compile extention with support for lz4 and igbinary if it helps:

-----



### Install lz4

```
brew install lz4
```


Check what folder it's installed to, in my case:

``` 
/opt/homebrew/Cellar/lz4/1.9.4
```


### Install igbinary

```
pecl install igbinary
```


```
 git clone https://github.com/phpredis/phpredis.git
 cd phpredis
 phpize
 ./configure --enable-redis-igbinary=yes --enable-redis-lzf=no --enable-redis-zstd=no --enable-redis-msgpack=no --enable-redis-lz4=yes --with-liblz4=/opt/homebrew/Cellar/lz4/1.9.4
 
```


I build from source as there is a bug in the current release available through pecl that is fixed in master a couple of weeks ago.

Read here:  https://github.com/phpredis/phpredis/issues/2452

**Comment by cyppe** [comment]
So it would be nice if you can either compile it with those settings.

Or load redis as a extention module in php.ini so we can easily change it to our own version.

Most long term solution would probably be to load as much as possible as extentions so we can load and unload anything we want so we don't need to get stuck while you consider some changes in core compiled extentions.

**Comment by cyppe** [comment]
Sorry for spamming, but one interesting point. 

Not sure why php 8.3 in herd uses so old version of phpredis?

**Fresh install of Herd and php 8.3**

```
Redis Support => enabled
Redis Version => 5.3.7
Redis Sentinel Version => 0.1
Available serializers => php, json
```


**Forge uses:**

```
Redis Version => 6.0.1
Redis Sentinel Version => 1.0
Available serializers => php, json, igbinary
Available compression => lzf, zstd, lz4
```


So can be worth doing something here for that reason too.

**Reply by cyppe** [reply]
Hope you have some more time to look into feature requests like this now when Herd for windows is out <3 

In meantime I had to compile my own binaries and replace the ones bundled with Herd. But it kind of defeats the purpose of Herd and also some glitches when loading  config for xdebug when Im doing it that way so a bit annoying.

**Comment by mpociot** [comment]
@cyppe PHP 8.3.7 on macOS now adds igbinary and lz4 support 🎉 

![image](https://github.com/beyondcode/herd-community/assets/804684/be7ff56e-8274-4184-bd88-b8b4f0ed8bf6)

**Comment by cyppe** [comment]
Thanks, thats so awesome! Will upgrade and try it.

**Reply by cyppe** [reply]
Works as expected, so now I finally dropped my custom compiled php binary and can enjoy the full Herd experience :) Thanks again.

**Comment by viccherubini** [comment]
Using Homebrew 4.4.19, here's how I got PHP 8.4.3 with the `igbinary`, `msgpack`, and `redis` extensions installed on a new MacBook M4:

```
brew install lz4 redis php
pecl install igbinary
pecl install msgpack
pecl install redis
```

In the last command, when prompted for `igbinary`, `msgpack`, and `lz4` support, you type `yes`. When prompted to use the system `lz4` library, do not type `yes` or `no`, but enter the path to where `lz4` was installed by Homebrew. As of the date of this post, that is `/opt/homebrew/Cellar/lz4/1.10.0`.

This avoids having to download the `redis` PECL package manually, and automatically places the `.so` file in the right location and updates the `php.ini` file to include it properly.

**Reply by mpociot** [reply]
This is now all part of Herd

**Reply by cyppe** [reply]
@mpociot A new member of our team uses Herd for windows and noticed that this extentions is not available there.  It also has a bit older version of phpredis.  

OSX:

:: herd php --ri redis

redis

Redis Support => enabled
**Redis Version => 6.1.0**
Redis Sentinel Version => 1.0
Available serializers => php, json, igbinary
Available compression => zstd, lz4


WINDOWS:

C:\Users\andrei\PhpstormProjects\motoaction_laravel8>  herd php --ri redis
redis
Redis Support => enabled
**Redis Version => 6.0.3-dev**
Redis Sentinel Version => 1.0
Available serializers => php

Is this something you might consider adding so Windows and OSX version is working the same here? Thanks.
